src/www/test.action.mjs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import url from 'url';
+import karma from 'karma';
+import puppeteer from 'puppeteer';
+import path from 'path';
+import {getRootDir} from '../build/get_root_dir.mjs';
+
+const KARMA_CONFIG_PATH = 'src/www/karma.conf.js';
+
+/**
+ * @description Runs the Karma tests against the web UI.
+ *
+ * @param {string[]} parameters
+ */
+export async function main() {
+  const runKarmaPromise = config =>
+    new Promise((resolve, reject) => {
+      new karma.Server(config, exitCode => {
+        if (exitCode !== 0) {
+          reject(exitCode);
+        }
+
+        resolve(exitCode);
+      }).start();
+    });
+
+  process.env.CHROMIUM_BIN = puppeteer.executablePath();
+
+  const config = await karma.config.parseConfig(path.resolve(getRootDir(), KARMA_CONFIG_PATH), null, {
+    promiseConfig: true,
+    throwErrors: true,
+  });
+
+  await runKarmaPromise(config);
```

Comment:
Ditch the `Promise` suffix and call it `runKarma`. It's an async method. And it should read like an action, not a noun.

Reply:
It's common in JS to append the type to the end of the name for clarity, especially when it's async like this, `Promise` tells you you'll need to `await` the result. 

It still reads like an action imo. `run` in the verb, `Karma` is the adjective, `Promise` is the noun. If I drop `Promise`, then `run` is the verb, `Karma` is the noun.

That said, I don't care all that much. It's right here in the same file and it's not like we're going to keep using it. I just wanted to explain my rationale.

Reply:
(btw async methods are just syntactical sugar on top of promises, they're interchangeable)

Reply:
I guess my point is that we don't append `Promise` to every `async` function name, even though they return a Promise. We just name them as regular methods.

Thanks for the change!

Reply:
FWIW, I have been trying to do that for things that aren't TS (see: https://github.com/Jigsaw-Code/outline-client/blob/8a4f4a19ff56b1f21a573f750e37018798acd8e5/src/build/webpack_promise.mjs#L3-L4)

Reply:
I'll update this too, IG